staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go

```diff
@@ -59,8 +59,8 @@ type Object interface {
 	SetFinalizers(finalizers []string)
 	GetOwnerReferences() []OwnerReference
 	SetOwnerReferences([]OwnerReference)
-	GetClusterName() string
-	SetClusterName(clusterName string)
+	GetZZZ_DeprecatedClusterName() string
+	SetZZZ_DeprecatedClusterName(clusterName string)
```

Comment:
I'm trying to decide if the interface change is better or worse than marking the interface methods deprecated (so linters would flag) and then dropping the methods in 1.25.

The struct field change, godoc, warning, and applyconfigurations changes were all much lower churn and those lgtm.

Reply:
The point is to give clients compile errors, and it turns out that most code (at least in k/k) interacts with metadata via this interface, so not changing it here would at least partly defeat the purpose.

Reply:
fair. the current state seems reasonable to me. will give a chance for others to chime in

Reply:
I'm ok with the interface change.